hummingbot/connector/exchange/btc_markets/btc_markets_exchange.py

```diff
@@ -0,0 +1,966 @@
+                                               self.current_timestamp,
+                                               tracked_order.client_order_id))
+                        tracked_order.cancelled_event.set()
+                        self.stop_tracking_order(order_msg['orderId'])
+                    elif tracked_order.is_failure:
+                        self.logger().info(f"Order {order_msg['orderId']} has failed according to order status API."
+                                           f"Reason: {btc_markets_utils.get_api_reason(order_msg['reason'])}")
+                        self.trigger_event(MarketEvent.OrderFailure,
+                                           MarketOrderFailureEvent(
+                                               self.current_timestamp,
+                                               tracked_order.client_order_id,
+                                               tracked_order.order_type
+                                           ))
+                        self.stop_tracking_order(order_msg['orderId'])
+                    else:
+                        self.logger().info(f"Order {order_msg['orderId']} is now {order_msg['status']}.")
+                else:
+                    self.logger().debug(f"Order {order_msg['orderId']} not currently tracked: ")
+
+        # Received trade match via orderBook WS message, so pass to process trade message
+        if order_msg["status"] in {'Partially Matched', 'Fully Matched'}:
+            await self._process_trade_message(order_msg)
+
+    async def _process_trade_message(self, trade_msg: Dict[str, Any]):
+        """
+        Updates in-flight order and trigger order filled event for trade message received. Triggers order completed
+        event if the total executed amount equals to the specified order amount.
+        """
+        if "orderId" not in trade_msg:
+            return
+        for order in self._in_flight_orders.values():
+            await order.get_exchange_order_id()
+        track_order = [o for o in self._in_flight_orders.values() if str(trade_msg["orderId"]) == o.exchange_order_id]
+        if not track_order:
+            return
+        tracked_order = track_order[0]
+        updated = tracked_order.update_with_trade_update(trade_msg)
+        if not updated:
+            return
+        for trade in trade_msg["trades"]:
+            self.trigger_event(
+                MarketEvent.OrderFilled,
+                OrderFilledEvent(
+                    self.current_timestamp,
+                    tracked_order.client_order_id,
+                    tracked_order.trading_pair,
+                    tracked_order.trade_type,
+                    tracked_order.order_type,
+                    Decimal(str(trade["price"])),
+                    Decimal(str(trade["volume"])),
+                    AddedToCostTradeFee(
+                        flat_fees=[TokenAmount(tracked_order.fee_asset, Decimal(str(trade["fee"])))]
+                    ),
```

Comment:
The code is using the `fee_asset` from the in flight order to create the fee object, but the `fee_asset` is never set, and hence it is None.